pkg/forwarder/blocked_endpoints.go

```diff
@@ -6,15 +6,22 @@
 package forwarder
 
 import (
+	"math"
 	"sync"
 	"time"
 )
 
 const (
-	blockInterval    time.Duration = 5 * time.Second
-	maxBlockInterval time.Duration = 30 * time.Second
+	// This controls how many retry interval ranges to step down for an endpoint
+	// upon success. Increasing this should only be considered when maxBackoffTime
+	// is particularly high or if our intake team is particularly confident.
+	recoveryInterval = 1
 )
 
+// This is the number of errors it will take to reach the maxBackoffTime. Our
+// blockedEndpoints circuit breaker uses this value as the maximum number of errors.
+var maxErrors = int(math.Floor(math.Log2(float64(maxBackoffTime)/float64(baseBackoffTime)))) + 1
```

Comment:
I think this should be abstracted into a function in `backoff.go` as this requires someone to know the implementation detail of it

Reply:
I'd really, really prefer not to add overhead to every circuit breaker action when this value is known at compile time and is imo just another implementation detail.